### PR TITLE
`Development`: Remove trailing slashes in URLs

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

- Extraction out of the Spring Boot 3 PR for easier review
- More sensitive route mapping in Spring Boot 3
- fixed two return types as they were mentioned by coderabbit

### Description

- Removed trailing slashes from all URLs
- Removed the test user prefix from admin users for one test

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Refactor**
	- Updated API endpoints across various services and tests for consistency by removing trailing slashes.
- **Tests**
	- Enhanced test configurations and updated annotations to improve reliability and consistency in testing environments.

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseIntegrationTest.java [hunk 1]
```diff
@@ -324,7 +324,7 @@
         }
 
         classExercise.setDueDate(ZonedDateTime.now().plusHours(12));
-        request.put("/api/modeling-exercises/", classExercise, HttpStatus.OK);
+        request.put("/api/modeling-exercises", classExercise, HttpStatus.OK);
```
[the remark this answers is not in the document]
Comment (julian-christl) [reply src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseIntegrationTest.java:327]: Fixed 21ce93202b7787c07a3a13ab04812c4ec174830c

File: src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseIntegrationTest.java [hunk 2]
```diff
@@ -595,7 +595,7 @@
         ExerciseGroup exerciseGroup = examUtilService.addExerciseGroupWithExamAndCourse(true);
         ModelingExercise modelingExercise = ModelingExerciseFactory.generateModelingExerciseForExam(DiagramType.ClassDiagram, exerciseGroup);
 
-        request.postWithResponseBody("/api/modeling-exercises/", invalidDates.applyTo(modelingExercise), ProgrammingExercise.class, HttpStatus.BAD_REQUEST);
+        request.postWithResponseBody("/api/modeling-exercises", invalidDates.applyTo(modelingExercise), ProgrammingExercise.class, HttpStatus.BAD_REQUEST);
```
[the remark this answers is not in the document]
Comment (julian-christl) [reply src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseIntegrationTest.java]: Fixed 21ce93202b7787c07a3a13ab04812c4ec174830c

**Review by mateusmm01 — APPROVED** [review]
code LGTM !

**Review by Strohgelaender — APPROVED** [review]
lgtm, did not find any missing cases.

**Comment by julian-christl** [comment]
Note: For E2E Tests only the same tets fail as on develop

**Pull request merged**